indicator-nara/031-berekening.Rmd

```diff
@@ -0,0 +1,707 @@
+
+
+list2env(
+  lapply(ruwetabellen_lijst, 
+         dbReadTable, conn = con,
+         guess_max = 300000), 
+  envir = .GlobalEnv)
+
+DBI::dbDisconnect(con)
+
+
+```
+
+```{r wegschrijven-brontabellen-naar-vc-formaat, eval=FALSE }
+for (i in seq(from = 1, to = 19)){
+  write_vc(get(sprintf("ruwedata%02d", i)), file.path("data", "local", sprintf("ruwedata%02d", i)), sorting = c("dag","meetpunt_import"), strict= FALSE)
+}
+
+```
+
+
+```{r inlezen-vc-tabellen, include=FALSE, eval=!file.exists(file.path("cache", "basisdata.rds" ))}
+# De data worden niet in één keer ingelezen.
+# De brondata zitten in 23 bestanden.
+# Elk van deze 23 bestanden is op zich al vrij omvangrijk.
+# Ze samenvoegen tot één bestand, zou kunnen leiden tot importproblemen.
+# De opdeling is ook wel praktisch als je met de data iets wil testen.
+
+ruwetabellen_lijst <- data.frame(bron = "ruwedata", doel = "ruwedata", c = seq(from = 1, to = 23))
+
+ruwetabellen_lijst <- ruwetabellen_lijst %>% mutate (naambrontabel = file.path("data", "local", paste0(bron,sprintf("%02d", c))), 
+                                                     naamdoeltabel = sprintf(paste0(doel,"%02d"), c)) 
+ruwetabellen_lijst_rds <- ruwetabellen_lijst %>% mutate(naamdoeltabel = file.path("cache", paste0(bron,sprintf("%02d", c), ".rds")), 
+                                                     naambrontabel = sprintf(paste0(doel,"%02d"), c)) 
+
+ruwetabellen_lijst <- setNames(ruwetabellen_lijst %>% dplyr::pull(naambrontabel), make.names(ruwetabellen_lijst %>% dplyr::pull(naamdoeltabel)))
+
+list2env(
+  lapply(ruwetabellen_lijst, 
+         read_vc), 
+  envir = .GlobalEnv)
+
+sapply(1:nrow(ruwetabellen_lijst_rds), function(i) {
+        filename <- ruwetabellen_lijst_rds$naamdoeltabel[i]
+        saveRDS(get(ruwetabellen_lijst_rds$naambrontabel[i]), filename)
+    })
+```
+```{r inlezen-vc-tabellen-uit-cache, include=FALSE}
+ruwetabellen_lijst <- data.frame(bron = "ruwedata", doel = "ruwedata", c = seq(from = 1, to = 23))
+
+ruwetabellen_lijst_rds <- ruwetabellen_lijst %>% mutate(naambrontabel = file.path("cache", paste0(bron,sprintf("%02d", c), ".rds")), 
+                                                     naamdoeltabel = sprintf(paste0(doel,"%02d"), c)) 
+ruwetabellen_lijst_rds <- setNames(ruwetabellen_lijst_rds %>% dplyr::pull(naambrontabel), make.names(ruwetabellen_lijst_rds %>% dplyr::pull(naamdoeltabel)))
+
+list2env(lapply(X = ruwetabellen_lijst_rds, FUN = readRDS), 
+  envir = .GlobalEnv)
+```
+
+```{r samenvoegen-bestanden, include=FALSE, eval=!file.exists(file.path("cache", "basisdata.rds" ))}
+#ik verkies dit in een aparte chunk te doen, omdat het zo mogelijk is om testdata samen te stellen, bijv. ik heb voor het uittesten van de methoden gewerkt met een testbestand van de eerste 5 ruwe tabellen.
+for (i in seq (from = 1, to = 23)){
+  if (i == 1) {
+    ruwedata <- get(sprintf("ruwedata%02d",i))
+  } else {
+    ruwedata <- bind_rows(ruwedata, get(sprintf("ruwedata%02d",i)))
```

Comment:
I think you can replace this entire for loop with the single line below. Untested.
```
map_df(sprintf("ruwedata%02d", 1:23), get)
```

Reply:
yes, it works !